src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -67,14 +89,11 @@ public ProxyGeneratorResult GenerateProxy(Type typeOfProxy, IEnumerable<Type> ad
                 return new ProxyGeneratorResult(DynamicProxyResources.ProxyIsSealedTypeMessage.FormatInvariant(typeOfProxy));
             }
 
+            Guard.AgainstNull(additionalInterfacesToImplement, "additionalInterfacesToImplement");
             GuardAgainstConstructorArgumentsForInterfaceType(typeOfProxy, argumentsForConstructor);
+            Guard.AgainstNull(fakeCallProcessorProvider, "fakeCallProcessorProvider");

```

Comment:
> Having guard clauses for arguments half way through methods is also a massive code smell.

The `GuardAgainstConstructorArgumentsForInterfaceType` was already in the middle, it's why I put the others there. I think it was an optimization to bail early if the type is a value type or is sealed. But it's not a very useful optimization, so I'll just put everything at the start of the method.